### ISSUE TITLE
`MediaType` is not exported to parent projecy

### ISSUE THREAD

**Issue by bambinoua** [dart-lang/http#873]
When I use `MultipartFile` constructors I could not specify the `contentType` property because `MediaType` from `http_parser` package is used internally in `multipart_file.dart` file and not exported into the target project.

**Comment by jamiesyme** [comment]
I just ran into this. Also see #342 and #296.

You can fix this by including the `http_parser` library (although I'm also curious if it's worth re-exporting `MediaType` as part of the `http` package).

```dart
import 'package:http_parser/http_parser.dart';
// or
import 'package:http_parser/http_parser.dart' show MediaType;
```

**Comment by bambinoua** [comment]
Yes. I did so but it is strange to import a whole additional package just for import the data type which is used by class from `http`, isn't it?

**Comment by Reprevise** [comment]
I'm curious if the Dart team would be interested in exporting MediaType from the `http` package. I added `DioMediaType` (type alias for http_parser's `MediaType`) which will be out in the next release of `dio`.

**Comment by brianquinlan** [comment]
@natebosch what do you think about this?

`MediaType` is exposed in the public `package:http` API for multipart uploads e.g.

```dart
Future<MultipartFile> multipartFileFromPath(String field, String filePath,
    {String? filename, MediaType? contentType});
```

My feeling is that we might not want to export types from other packages. It seems like we'd be moving down a slippery slope.

**Comment by Reprevise** [comment]
I dont think it's good to have users add what should be a transitive dependency just to use a feature of the package. I'm not sure what the slippery slope aspect is. The http_parser package is maintained by the Dart team.

**Comment by brianquinlan** [comment]
> I dont think it's good to have users add what should be a transitive dependency just to use a feature of the package. I'm not sure what the slippery slope aspect is. The http_parser package is maintained by the Dart team.

The slippery slope aspect is would we do this for all packages and for all types used in other packages? For example, if we had 3 packages that use `PriorityQueue` from `package:collection` then would all of them export it?

**Comment by Reprevise** [comment]
> For example, if we had 3 packages that use PriorityQueue from package:collection then would all of them export it?

If `PriorityQueue` was used in a public API, then I'd expect the type to be available to have it without having to import another package and add it to my pubspec.yaml (you get a warning for `import` statements for packages that are not direct dependencies). Or you could _rely_ on the fact that package:collection is a pretty common direct dependency already (though I'd still argue for re-export in that case personally).

I'm willing to bet that 95% of applications don't have `http_parser` as a _direct_ dependency. You _can_ still import it while it's a transitive dependency but you'd get a warning as mentioned before.
